_posts/2021-11-11-java-release-notes.md

```diff
@@ -0,0 +1,691 @@
+} catch (IOException e) {
+    e.printStackTrace();
+}finally {
+    try {
+        br.close();
+    } catch (IOException e) {
+        e.printStackTrace();
+    }
+}
+```
+After calling upon method on autoclosable interface, we would, in `finally` block, call `close()` method. The `finally` block ensures that the reader is always properly closed.
+
+#### Improvement on Autoclosable
+Java 8 introduced `try with resource` feature that enables us to declare our resource inside `try` definition. This will ensure that our closable is closed without using `finally`. Let us take a look at some example of using the `BufferedReader` to read string:
+```java
+final BufferedReader br3 = new BufferedReader(new StringReader("Hello world example3!"));
+try(BufferedReader reader = br3){
+    System.out.println(reader.readLine());
+}catch (IOException e){
+    System.out.println("Error happened!");
+}
+```
+Inside `try` definition, we assigned our previously created reader to the new variable. Now we know that our reader will get closed every time.
+#### Improvement upon Try-with-resource
+How can we improve [the example](#improvement-on-autoclosable) from above? 
+
+Well, defining a variable in one place only to reassign it to another inside `try` definition is a little bit of pain in the eye.
+With Java 9, we got changes with which we can avoid that scenario. We will be looking into the same example of reading input string through `BufferedReader`:
+```java
+final BufferedReader br2 = new BufferedReader(new StringReader("Hello world example2!"));
+try(br2){
+    System.out.println(br2.readLine());
+}
+catch (IOException e){
+    System.out.println("Error happened!");
+}
+```
+Now we don't need to create a new variable only to autoclose it inside the `try-catch` block. We are using the same one from the original definition.
+
+To read more about `try-with-resources` feature please refer to [oracle page.](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)
+### Diamond Syntax with Inner Anonymous Classes
+#### What Java 9 Fixed
+Before Java 9 we couldn't use a diamond operator inside inner anonymus class.
+
+For our example, we will create an abstract class. It has only one method that appends two strings with `-` between them. 
+Since this is an abstract class, we will use the anonymous class for providing the implementation for the `append()` method:
+```java
+public static void main(String[] args) {
+    AppendingString<String> appending = new AppendingString<>() {
+        @Override
+        public String append(String a, String b) {
+            return new StringBuilder(a).append("-").append(b).toString();
+        }
+    };
+
+    String result = appending.append("Reflectoring", "Blog");
+    System.out.println(result);
+}
+
+public abstract static class AppendingString<T>{
+    public abstract T append(String a, String b);
+}
+```
+We are using the diamond operator to tell our method which type we expect.
```

Comment:
I had to search through the code a bit to find the place where we use the diamond operator. Also, we're not exactly using it to tell the method which type to expect, since there is no type in the diamond operator!